tools/azsdk-cli/docs/new-tool.md

```diff
@@ -18,6 +18,7 @@ Help me create a new tool using #new-tool.md as a reference
    * [Registration and Testing](#registration-and-testing)
    * [Required Tool Conventions](#required-tool-conventions)
    * [Common Patterns and Anti-patterns](#common-patterns-and-anti-patterns)
+   * [CLI Command Hierarchy](#cli-command-hierarchy)
```

Comment:
nit: consider moving this before the step-by-step guide so that a new user gets the tool architecture and the cli commands